Update how-to-select-technology.md (#95)
bcgov/bcdg: docs/choosing-technology/how-to-select-technology.md

# Choosing project technology 

> Every team and ministry has different needs, but in CSBC we are working towards finding, supporting and improving standard tools to be widely adopted across our technology ecosystem. Choosing the widely adopted technology makes it easier for us to build the foundation of excellent user experience, and support applications in the long term. 

Reviewer: ```suggestion
Every team and ministry has different needs, but in Connected Services B.C. (CSBC) we are working towards finding, supporting and improving standard tools to be widely adopted across our technology ecosystem. Choosing the widely adopted technology makes it easier for us to build the foundation of excellent user experience, and support applications in the long term. 
```

> We offer many options within B.C. Government systems. The tables below show a cross-section of some technology that developers should try to use. Users can click each option to get more information about it.  

Reviewer: ```suggestion
We offer many options within B.C. government systems. The tables below show a cross-section of some technology that developers should try to use. Users can click each option to get more information about it.  
```

### Development Frameworks
Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. 

|Type| Documentation |
|----| ------|
|Programming languages|[Python](https://github.com/bcgov?q=&type=all&language=python&sort=), [Javascript](https://github.com/bcgov?q=&type=all&language=javascript&sort=), [TypeScript](https://github.com/bcgov?q=&type=all&language=typescript&sort=), [Java](https://github.com/bcgov?q=&type=all&language=java&sort=), [R](https://github.com/bcgov?q=&type=all&language=r&sort=), [C#](https://github.com/bcgov?q=&type=all&language=c%23&sort=), [PHP](https://github.com/bcgov?q=&type=all&language=php&sort=)
|Back-end| [Django](https://www.djangoproject.com/), [Flask](https://flask.palletsprojects.com/en/2.2.x/), [Express](https://expressjs.com/), [NestJS](https://nestjs.com/), [Spring Boot](https://spring.io/projects/spring-boot)[.NET Core](https://dotnet.microsoft.com/en-us/apps/aspnet), [Laravel](https://laravel.com/)
|Front-end| [React](https://reactjs.org/), [Vue](https://vuejs.org/)|

> ### Application Development Tooling

Reviewer: ```suggestion
### Application development tooling
```

> Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for BC Government related support channels, if applicable.

Reviewer: ```suggestion
Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for B.C. government related support channels, if applicable.
```

| Secrets Management |[Vault*](/docs/default/component/platform-developer-docs/docs/secrets-management/vault-getting-started-guide/)| [OpenShift-howto-vault](https://teams.microsoft.com/l/channel/19%3A0ed86e1a6dd74b62a7d965f1a6954c56%40thread.tacv2/OpenShift-howto-vault?groupId=a80418da-c27b-406e-89ab-7695b61924d8&tenantId=6fdb5200-3d0d-4a8a-b036-d3685e359adc)
| Status & Alerts | Known gap
| Source Control & Collaboration | [GitHub](https://developer.gov.bc.ca/docs/default/component/bc-developer-guide/use-github-in-bcgov/quick-reference/) | [DevX Team Support Requests](https://citz-do.atlassian.net/servicedesk/customer/portal/2), [GitHub-howto](https://teams.microsoft.com/l/channel/19%3Aae5f84cbecd3430793841f3944632767%40thread.tacv2/GitHub-howto?groupId=a80418da-c27b-406e-89ab-7695b61924d8&tenantId=6fdb5200-3d0d-4a8a-b036-d3685e359adc)

> \* these tools are currently only corporately available only within the Private Cloud ecosystem

Reviewer 2: ```suggestion
\* these tools are currently only available within the Private Cloud ecosystem
```

Removed `only corporately` - reduced to one `only` in the sentence. Don't think `corporately` is needed but fine for it to be put back in the sentence

> ### Collaboration & Knowledge Sharing

Reviewer: ```suggestion
### Collaboration and knowledge sharing
```

> Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for BC Government related support channels, if applicable.

Reviewer: ```suggestion
Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for B.C. government related support channels, if applicable.
```

> Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for B.C. government related support channels, if applicable.

Reviewer 2: ```suggestion
Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. 
```
Column 3 (support isn't used in the table below). Removed reference to it

> |Type| Documentation | Support |

Reviewer 2: The Support column isn't populated in this table. Maybe remove it since it is not used?

|----| ------|------|
| Chat & Video | [MS Teams](https://www.microsoft.com/en-ca/microsoft-teams/group-chat-software)

> | Design & mockups | Figma

Reviewer 2: Can we add a link to Figma? Or how to request it?

| Technical Documentation | [DevHub Techdocs](https://developer.gov.bc.ca/docs/default/component/bc-developer-guide/content-partner-guide/) 
| Technical Q&A | [GitHub Discussions](https://github.com/features/discussions), 
|Ticketing and backlog| [JIRA](https://www.atlassian.com/software/jira) 

### Common Components

> Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for BC Government related support channels, if applicable.

Reviewer: ```suggestion
Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for B.C. government related support channels, if applicable.
```

> Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for B.C. government related support channels, if applicable.

Reviewer 2: ```suggestion
Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. 
```
Column 3 (support isn't used in the table below). Removed reference to it

|Type| Documentation | Support |

> |----| ------|------|

Reviewer 2: The Support column isn't populated in this table. Maybe remove it since it is not used?

Reviewer: I added a few (but there are empties too) - I did remove the ones where none were available in the latest PR update.

| Design System | [Design System](https://digital.gov.bc.ca/bcgov-common-components/design-system/)
| Document Generation| [Common Document Generation Service](https://digital.gov.bc.ca/bcgov-common-components/common-document-generation-service/)
|Email generation| [Common Hosted Email Service](https://digital.gov.bc.ca/bcgov-common-components/common-hosted-email-service/)
|Forms | [Common Hosted Forms Service](https://digital.gov.bc.ca/bcgov-common-components/common-hosted-form-service/)
| Grants | Unity
| Identity verification | [Digital Trust](https://digital.gov.bc.ca/digital-trust/home/)
|Maps | [Simple Map Kit](https://digital.gov.bc.ca/bcgov-common-components/simple-map-kit/)
| Notifications | Notifications Team
| Object Management | [Common Object Management Service](https://digital.gov.bc.ca/bcgov-common-components/common-object-management-service/)
| Pay Authorizization | Pay Auth
| Permitting | Permit Connect Navigator Service (PCNS)
|Route planning | [BC Route Planner](https://digital.gov.bc.ca/bcgov-common-components/bc-route-planner/)
| Single sign on | [Keycloak SSO](docs/default/component/css-docs)
| Workflow | BC Gov managed n8n

> 

Reviewer 2: Are there links for some of these other services, like Pay Auth and the Notifications Team?

Author: My intention is to create a bit of a "backlog" of things we need to map out - so totally ok from my perspective to leave things blank that are still settling as an excuse for Charles to engage with them to bring them on as content partners when they're ready :)

Reviewer 2: Do they have a repo, JSM, other docs etc we can point people to in the meantime?

### Integration and data

> Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for BC Government related support channels, if applicable.

Reviewer: ```suggestion
Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for B.C. government related support channels, if applicable.
```

> Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for B.C. government related support channels, if applicable.

Reviewer 2: ```suggestion
Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. 
```
Column 3 (support isn't used in the table below). Removed reference to it

> |Type| Documentation | Support |

Reviewer 2: The Support column isn't populated in this table. Maybe remove it since it is not used?

|----| ------|------|
|API Gateway & Secure Data Exchange| [API Gateway](https://digital.gov.bc.ca/bcgov-common-components/api-program-services/)
| Data Catalogue | [BC Data Catalogue](https://catalogue.data.gov.bc.ca)
| Data & Analytics Platform | [Microsoft Fabric](https://developer.gov.bc.ca/docs/default/component/public-cloud-techdocs/azure/azure-services/external-fabric/) | 
|Middleware and databases| PostgreSQL, MySQL, MongoDB, Kafka, Redis, NATS                                                                          

### AI Services

> Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for BC Government related support channels, if applicable.

Reviewer: ```suggestion
Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for B.C. government related support channels, if applicable.
```

> Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. Column 3 lists information for B.C. government related support channels, if applicable.

Reviewer 2: ```suggestion
Column 1 lists the technology type. Column 2 lists the documentation for the choices under that type. 
```
Column 3 (support isn't used in the table below). Removed reference to it

|Type| Documentation | Support |

> |----| ------|------|

Reviewer 2: The Support column isn't populated in this table. Maybe remove it since it is not used?

| Secure AI Models as a service | AI Services Hub
| Forms validation | AI Forms Assist
| Chatbot | AI Chatbot

> 

Reviewer 2: What are these services? I think we need to provide some links so people can use them/find info on them easily.

Reviewer: In latest PR i removed (aka hide)  these resources for now, I couldn't find any official documentation, but I did find the official BC doc regarding AI.

Reviewer: Hid for now. Added official BC website with regards to AI services for now.
